### DISCUSSION TITLE
Core, dispensable, and private genome

### DISCUSSION THREAD

**Discussion by beantkapoor786** [maize-genetics/phg_v2#200] [Q&A]
Once I have gone through the PHG build and load module, how can I calculate the core, dispensable, and private genome percentage?

**Comment by btmonier** [comment]
If you are talking about unique and shared haplotypes across the genome, one option would be to use rPHG2 and generate a [haplotype ID matrix](https://rphg2.maizegenetics.net/articles/rPHG2.html#haplotype-ids) to summarize unique IDs across each column (reference ranges):

``` r
library(rPHG2)

phgLib <- "path/to/your/phgv2/lib/dir"
initPhg(phgLib)

hvcfFiles <- hvcfDir |> list.files(pattern = ".h.vcf")

graph <- hvcfFiles |> 
    PhgLocalCon() |> 
    buildHaplotypeGraph()

hapIds <- graph |> readHapIds()

hapProfile <- data.frame(
    ref_range  = hapIds |> colnames(),
    n_uniq_ids = hapIds |> apply(2, \(it) it |> unique() |> length())
)
```

**Reply by beantkapoor786** [reply]
Thank you, will try this.
